src/commands/dev/index.js

```diff
@@ -82,7 +91,30 @@ const startFrameworkServer = async function ({ settings, log, exit }) {
   })
 
   try {
-    const open = await waitPort({ port: settings.frameworkPort, output: 'silent', timeout: FRAMEWORK_PORT_TIMEOUT })
+    const open = await waitPort({
+      port: settings.frameworkPort,
+      output: 'silent',
+      timeout: FRAMEWORK_PORT_TIMEOUT,
+    })
+
+    const waitForServerToRespond = async () => {
+      try {
+        await fetch(`http://localhost:${settings.frameworkPort}`, {
+          method: 'HEAD',
+          timeout: SERVER_POLL_REQUEST_TIMEOUT,
```

Comment:
I'm not sure the `fetch` API allows passing a timeout.

We can consider using https://github.com/sindresorhus/got but that's out of scope for this PR.

Reply:
> Hi @eduardoboucas - I like the approach in this PR!
> I have 2 questions/suggestions:
> 
> 1. Can we verify a few other framework servers accept `HEAD` requests?
> 2. If `waitFor` times out and `open` is `true`, how about we print a warning instead of failing?

Both great suggestions, thanks @erezrokah!

1. I tested a few, happy to test a few more. Would you feel more comfortable if we used a `GET` request instead?
2. I was actually experimenting with this locally. Since you brought it up, we should definitely do it.

Reply:
I think it exists in the version of `node-fetch` we're using, but it was [removed in version 3](https://github.com/node-fetch/node-fetch/blob/cbd4d895767e33491219e40013dd8daa5c7ac024/docs/v3-UPGRADE-GUIDE.md#the-timeout-option-was-removed). Definitely something we need to take into account if/when upgrading.

Reply:
Oh that's very helpful to know thanks!